modules/battenberg/1.2/battenberg.smk

```diff
@@ -0,0 +1,297 @@
+        probloci =  CFG["dirs"]["inputs"] + "reference/{genome_build}/probloci.txt.gz",
+        battenberg_wgs_replic_correction = directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_wgs_replic_correction_1000g_v3"),
+        battenberg_gc_correction = directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_wgs_gc_correction_1000g_v3"),  
+        genomesloci = directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_1000genomesloci2012_v3")
+    params:
+        url = "https://www.bcgsc.ca/downloads/morinlab/reference",
+        alt_build = lambda w: VERSION_MAP[w.genome_build],
+        folder = CFG["dirs"]["inputs"] + "reference/{genome_build}",
+        build = "{genome_build}",
+        PATH = CFG['inputs']['src_dir']
+    resources:
+        **CFG["resources"]["reference"]
+    threads:
+        CFG["threads"]["reference"]
+    shell:
+        op.as_one_line("""
+        wget -qO-  {params.url}/battenberg_impute_{params.alt_build}.tar.gz  |
+        tar -xvz > {output.battenberg_impute} -C {params.folder}
+        &&
+        wget -qO- {params.url}/battenberg_{params.alt_build}_gc_correction.tar.gz  |
+        tar -xvz > {output.battenberg_gc_correction} -C {params.folder}
+        &&
+        wget -qO- {params.url}/battenberg_1000genomesloci_{params.alt_build}.tar.gz | 
+        tar -xvz > {output.genomesloci} -C {params.folder}
+        &&
+        wget -O {output.impute_info} {params.url}/impute_info_{params.alt_build}.txt
+        &&
+        python {params.PATH}/reference_correction.py {params.build}
+        &&
+        wget -qO-  {params.url}/battenberg_{params.alt_build}_replic_correction.tar.gz |
+        tar -xvz > {output.battenberg_wgs_replic_correction} -C {params.folder}
+        &&
+        wget -O {output.probloci} {params.url}/probloci_{params.alt_build}.txt.gz
+        
+        """)
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _battenberg_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.crai"
+    group: "setup_run"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bam + ".bai", output.bai)
+        op.absolute_symlink(input.bam + ".bai", output.crai)
+
+# Installs the Battenberg R dependencies and associated software (impute2, alleleCounter)
+# Currently I think this rule has to be run twice for it to work properly because the conda environment is created here. 
+# I am open to suggestions for how to get around this.
+rule _install_battenberg:
+    output:
+        complete = "config/envs/battenberg_dependencies_installed.success"
```

Comment:
can you change it to touch the file in the CFG["dirs"]["input"]instead of config/envs?

Reply:
done

Reply:
Thanks! Have you tried running it, or just modified the line? The `CFG["dirs"]["input"] + "/battenberg_dependencies_installed.success"` will create path with `//` in it, so it is best to avoid whenever possible. This file is also used as input in the subsequent rule, did you look at that?